# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditUtils.h
# Commit: 5bacd687fa98
# Full Hash: 5bacd687fa98b0e9a5dea9944f360fdc5a6770e6
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-01-17 16:13:02
# Regressor Bug: 1807829
# File Overlap Count: 2
# Description:
#   Bug 1807829 - Make `HTMLEditor::CreateStyleForInsertText` use same logic as `HTMLEditor::SetInlinePropertiesAsSubAction` to apply new style r=m_kato
#   
#   The problem in Yahoo mail is, when you apply new font size to:
#   ```
#   <div><font size="7">{}<br></font></div>
# ==============================================================================

diff -r a7465d2d89ba -r 5bacd687fa98 editor/libeditor/HTMLEditUtils.h
--- a/editor/libeditor/HTMLEditUtils.h	Mon Jan 16 23:14:12 2023 +0000
+++ b/editor/libeditor/HTMLEditUtils.h	Tue Jan 17 05:26:14 2023 +0000
@@ -2002,6 +2002,14 @@
       const Element& aEditingHost);
 
   /**
+   * GetBetterCaretPositionToInsertText() returns better point to put caret
+   * if aPoint is near a text node or in non-container node.
+   */
+  template <typename EditorDOMPointType, typename EditorDOMPointTypeInput>
+  static EditorDOMPointType GetBetterCaretPositionToInsertText(
+      const EditorDOMPointTypeInput& aPoint, const Element& aEditingHost);
+
+  /**
    * ComputePointToPutCaretInElementIfOutside() returns a good point in aElement
    * to put caret if aCurrentPoint is outside of aElement.
    *